Sources/_StringProcessing/Engine/StringProcessor.swift

```diff
@@ -0,0 +1,31 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2021-2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+public // For prototypes
```

Comment:
Whatever's public in `_StringProcessing` is public to the user. Why do we want to expose `Program` or `MEProgram`?

Reply:
It's because we're restricted to only 2 modules here. The engine should be its own module, internal to everything else, so that it can have API that's usable by prototypes and `_StringProcessing`. But since we have to slurp everything up into `_StringProcessing`, then this needs to service the prototypes too.

We can drop the prototypes support at some point, at which point this would be dropped. We have loads of public entities that we know we aren't going to ship as public already, so there will be an audit / purge. I can denote it with a FIXME comment or something to make it easier to find in the future.

Reply:
Would something like `@_spi(PrototypeSupport)` work?